features/256-design-for-user-attribute-operations-in-next/api.md

```diff
@@ -0,0 +1,178 @@
+# Design for user attribute operations in Next
+
+Save user attributes in records gear directly (or auth gear), this feature spec is aimed to solve skygear next has to introduce new API to get user attributes if user attributes are not stored in one place.
+
+This change would solve the above question and bring some benefits:
+
+- auth data stored in one place is a preferred design.
+- user attribute can be configured with ACL.
+- user attribute can be used by record query.
+
+## Predefined user attributes
+
+Predefined user attributes are:
+
+- `last_seen_at`
+- `last_login_at`
+- `disabled`
+- `disabled_message`
+- `disabled_expiry`
+- `verified`
+- `xxx_verified` (`xxx_verified` would change when server configuration is changed.)
+
+## If record gear is configured
+
+If record gear is configured, predefined user attributes will be saved as columns of `_record__user` table, and a developer can use a rich API set which is provided by record gear to access user record.
+
+## If record gear is not configured
+
+If record gear is not configured, predefined user attributes will be saved as columns of `_auth_user_profile`. Auth gear will provide limited API to support user attributes access operations (query will not be provided).
```

Comment:
Will ACL provided?

And will their listing of user supported in SDK?

Reply:
1. Default `record` attributes should be included, like "_access", "_created_at", "_updated_at", ...
2. As discuss before, auth gear won't support listing.

Reply:
I mean are we going to support ACL if recrod grear is not configured. I think the answer is no. And please make it explicit no in the feature spec.

2. Yes, we don't support listing. Than we should explicit state that in the spec.